### PR TITLE
Village bug fix

### LINKED ISSUE #813: Bug: Unable to Enter Settlements after some time
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

After a few minutes of fast forward, the clients party can no longer enter settlements
Find out why and fix this issue

## Intended Design

N/A

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### LINKED ISSUE #818: Bug: Choosing "Wait for some time" in Village Leaves Village
# For New Contributors
[How to contribute](https://github.com/Bannerlord-Coop-Team/BannerlordCoop/wiki/Getting-Started-as-a-Contributor)

## Description

Choosing "Wait for some time" in a village will have the party leave the village while the UI is gone. Implement a fix for this.

## Intended Design

N/A

## Location

Create a branch based from [development](https://github.com/Bannerlord-Coop-Team/BannerlordCoop)

## Related Issues

N/A

## Requirements

N/A

## Additional information

N/A

## Definition of Done
- [ ] Class level comments exist for all new classes.
- [ ] XUnit tests exist for every method that does not require the game to be ran.

### PR DESCRIPTION
**Pull request by georgyrudnev**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] All new classes have class-level documentation comments, if there are any at all
- [ ] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation update
- [ ] Other... Please describe:

## What is the current behavior?

UI disappears when entering settlement and clicking "Wait here for some time"

## What is the new behavior?

Resolves #818

### CODE REVIEW

File: source/Common/Messaging/MessageBroker.cs
Comment (garrettluskey) [inline source/Common/Messaging/MessageBroker.cs:1]: Revert this file to what is on development

File: source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs [hunk 1]
```diff
@@ -106,7 +106,7 @@ public void StartPlayerSettlementEncounter(string partyId, string settlementId)
             return;
         }
 
-        if (PlayerEncounter.Current is not null) return;
+       // if (PlayerEncounter.Current is not null) return;
```
Comment (garrettluskey) [inline source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs:109]: does this need to be commented out?

File: source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs [hunk 2]
```diff
@@ -89,13 +89,13 @@ public void StartPlayerSettlementEncounter(string partyId, string settlementId)
     {
         if (objectManager.TryGetObject(partyId, out MobileParty mobileParty) == false)
         {
-            Logger.Error("PartyId not found: {id}", partyId);
+             Logger.Error("PartyId not found: {id}", partyId);
```
Comment (garrettluskey) [inline source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs:92]: fix spacing please

File: .gitignore [hunk 3]
```diff
@@ -549,4 +549,4 @@ source/.idea
 !*Coop.csproj.user
 !*MissionTestMod.csproj.user
 # CodeRush
-/source/.cr/personal
\ No newline at end of file
+/source/.cr/personal
```
Comment (garrettluskey) [inline .gitignore:552]:
```suggestion
**/.cr/**
```

**Review by garrettluskey — CHANGES_REQUESTED** [review]

File: source/Common/Messaging/MessageBroker.cs
Comment (georgyrudnev) [reply source/Common/Messaging/MessageBroker.cs:1]: This one seems to be part of the reason why the bug is ocurring. I changed it back how it was before October.

File: source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs (hunk 1 above)
Comment (georgyrudnev) [reply source/GameInterface/Services/MobileParties/Interfaces/MobilePartyInterface.cs:109]: I tested it both ways and it also seems to be related to the bug.

**Comment by garrettluskey** [comment]
This was fixed by #1028

**Pull request closed without merging**